From 9b10706f145dc4eb99d91bb2c04f551f7a55ea95 Mon Sep 17 00:00:00 2001
From: Alexey Andriyanov <alan@al-an.info>
Date: Wed, 21 May 2014 11:49:09 +0400
Subject: [PATCH 4/6] log_message: emit -Wformat= compiler warnings

There could be (and actually are) situations when the format string and
the arguments list passed to the log_message() are inconsistent or mistyped.
The compiler did not show any warnings because the vsnprintf was called
indirectly.
---
 lib/logger.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/lib/logger.h b/lib/logger.h
index 4796cf5..c1db5c4 100644
--- a/lib/logger.h
+++ b/lib/logger.h
@@ -23,7 +23,15 @@
 #ifndef _LOGGER_H
 #define _LOGGER_H
 
+#include <stdio.h>
+
 void enable_console_log(void);
-void log_message(int priority, char* format, ...);
+void log_message(int priority, const char* format, ...);
+
+/* wrapper around the real log_message() to emit -Wformat= warnings */
+#define log_message(priority, format, ...) do {\
+	if (0) fprintf (NULL, format, ##__VA_ARGS__); \
+	(log_message) (priority, format, ##__VA_ARGS__); \
+} while (0)
 
 #endif
-- 
1.8.3.2

